### PR TITLE
Run AGP tests with versions 3.5.2 and 3.6.0-beta03

### PR DESCRIPTION
**Pull request by lptr**

### CODE REVIEW

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerSmokeTest.groovy [hunk 1]
```diff
@@ -68,13 +65,6 @@ class AndroidSantaTrackerSmokeTest extends AbstractSmokeTest {
...
-            "Type 'DependencyTask': field 'configurations' without corresponding getter has been annotated with @Input. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.",
-            "Type 'DependencyTask': field 'outputDir' without corresponding getter has been annotated with @OutputDirectory. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.",
-            "Type 'DependencyTask': field 'outputFile' without corresponding getter has been annotated with @OutputFile. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.",
-            "Type 'LicensesTask': field 'dependenciesJson' without corresponding getter has been annotated with @InputFile. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.",
```
Comment (wolfs) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerSmokeTest.groovy]: Didn't we remove the `LicenseTask` from the `santa-tracker` build: https://github.com/gradle/santa-tracker-android/commit/d4d6b2470f9752f19198341ccd4e6a9719b82832

That might also explain why the `DependencyTask` doesn't show up any more.

Note that the plugin is not part of AGP, so I think it is fine.

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerSmokeTest.groovy [hunk 2]
```diff
@@ -50,9 +50,6 @@ class AndroidSantaTrackerSmokeTest extends AbstractSmokeTest {
 
         expect:
         expectDeprecationWarnings(result,
-            "Property 'excludeListProvider' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.",
```
Comment (wolfs) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerSmokeTest.groovy]: Why don't we have a type for this `Property`? Where is it declared (has it been declared)?

Comment (lptr) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerSmokeTest.groovy]: This message is (was) printed right underneath the name of the task that has the property. It's not ideal, as you still need to figure out the type of the task (and which plugin it belongs to), but any other approach I tried to display the type name led to too verbose outputs.

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerSmokeTest.groovy (hunk 1 above)
Comment (lptr) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerSmokeTest.groovy]: I've submitted a PR for the licenses plugin here: https://github.com/google/play-services-plugins/pull/111

**Pull request merged**